set_1_basics/challenge_6_break_repeating_key_xor/decode_base64.cpp

```diff
@@ -68,16 +64,14 @@ std::istream & operator>>(std::istream & input,
 
         value = decode_base64::convertFromBase64(base64Digit);
         static const size_t s_numRemainingBitsInSecondOctet =
-                decode_base64::s_numBitsInOctet - s_numBitsInSecondOctet;
+            decode_base64::s_numBitsInOctet - s_numBitsInSecondOctet;
         static const size_t s_numBitsInThirdOctet =
-                decode_base64::s_numBitsInBase64Digit
-                - s_numRemainingBitsInSecondOctet;
+            decode_base64::s_numBitsInBase64Digit
+            - s_numRemainingBitsInSecondOctet;
         manipulator.d_output[1].get() += value >> s_numBitsInThirdOctet;
-        manipulator.d_output[2] = (value & ((1u << s_numBitsInThirdOctet) - 1))
-                                  << (
-                                        decode_base64::s_numBitsInOctet
-                                        - s_numBitsInThirdOctet
-                                     );
+        manipulator.d_output[2] =
+            (value & ((1u << s_numBitsInThirdOctet) - 1))
+            << (decode_base64::s_numBitsInOctet - s_numBitsInThirdOctet);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Do not access "s_numBitsInOctet" with derived type "decode_base64". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3252)